src/shared_memory_manager.cc

```diff
@@ -176,55 +178,63 @@ CloseSharedMemoryRegion(int shm_fd)
             std::string(std::strerror(errno)))
             .c_str());
   }
+#endif
 
   return nullptr;
 }
 
 TRITONSERVER_Error*
-UnmapSharedMemory(void* mapped_addr, size_t byte_size)
+SharedMemoryManager::MapSharedMemory(
+    ShmFile* shm_file, const size_t offset, const size_t byte_size,
+    void** mapped_addr)
 {
-  int status = munmap(mapped_addr, byte_size);
-  if (status == -1) {
+#ifdef _WIN32
+  // The MapViewOfFile function takes a high-order and low-order DWORD (4 bytes
+  // each) for offset. 'size_t' can either be 4 or 8 bytes depending on the
+  // operating system. To handle both cases agnostically, we cast 'offset' to
+  // uint64 to ensure we have a known size and enough space to perform our
+  // logical operations.
+  uint64_t upperbound_offset = (uint64_t)offset;
+  DWORD high_order_offset = (upperbound_offset >> 32) & 0xFFFFFFFF;
+  DWORD low_order_offset = upperbound_offset & 0xFFFFFFFF;
+  // map shared memory to process address space
+  *mapped_addr = MapViewOfFile(
+      shm_file->shm_handle_,  // handle to map object
+      FILE_MAP_ALL_ACCESS,    // read/write permission
+      high_order_offset,      // offset (high-order DWORD)
+      low_order_offset,       // offset (low-order DWORD)
+      byte_size);
+
+  if (*mapped_addr == NULL) {
+    CloseSharedMemoryRegion(shm_handle);
     return TRITONSERVER_ErrorNew(
         TRITONSERVER_ERROR_INTERNAL,
         std::string(
-            "unable to munmap shared memory region, errno: " +
-            std::string(std::strerror(errno)))
+            "unable to process address space, error code: " + GetWindowsError())
             .c_str());
   }
-
-  return nullptr;
-}
-
-#ifdef TRITON_ENABLE_GPU
-TRITONSERVER_Error*
-OpenCudaIPCRegion(
-    const cudaIpcMemHandle_t* cuda_shm_handle, void** data_ptr, int device_id)
-{
-  // Set to device curres
-  cudaSetDevice(device_id);
-
-  // Open CUDA IPC handle and read data from it
-  cudaError_t err = cudaIpcOpenMemHandle(
-      data_ptr, *cuda_shm_handle, cudaIpcMemLazyEnablePeerAccess);
-  if (err != cudaSuccess) {
+#else
+  // map shared memory to process address space
+  *mapped_addr = mmap(
+      NULL, byte_size, PROT_WRITE | PROT_READ, MAP_SHARED, shm_file->shm_fd_,
+      offset);
+  if (*mapped_addr == MAP_FAILED) {
     return TRITONSERVER_ErrorNew(
         TRITONSERVER_ERROR_INTERNAL, std::string(
-                                         "failed to open CUDA IPC handle: " +
-                                         std::string(cudaGetErrorString(err)))
+                                         "unable to process address space " +
+                                         std::string(std::strerror(errno)))
                                          .c_str());
   }
-
+#endif
   return nullptr;
 }
-#endif  // TRITON_ENABLE_GPU
-
-}  // namespace
 
 SharedMemoryManager::~SharedMemoryManager()
 {
   UnregisterAll(TRITONSERVER_MEMORY_CPU);
+#ifndef _WIN32
```

Comment:
nit: I don't think you need this ifdef since no CUDA shared memory will be registered on Windows in the first place.

Reply:
Placed due to an abundance of caution. Removed without testing error.